internal/pkg/service/buffer/dependencies/dependencies.go

```diff
@@ -75,7 +75,10 @@ func NewServiceDeps(
 
 	// Create etcd client
 	etcdClient, err := etcdclient.New(
-		proc.Ctx(),
+		// Use separated context.
+		// Graceful shutdown is handled by Process.OnShutdown bellow.
+		// During the shutdown it is necessary to complete some etcd operations.
+		context.Background(),
```

Comment:
Ked som skusal worker este manualne, tak som narazil na tento problem.

Ked sa worker procesu posle graceful shutdown signal `SIGTERM`.

Tak sa ukonci hlavny context, `proc.Ctx()` a etcd klient sa odpoji.
Nahodne bud stihne dokoncit requesty alebo nie.

Preto som vytvoril samostatny kontext,
shutdown je riadeny cez `OnShutdown` nizsie.

Pripominam, ze shutdown je LIFO, teda operacie, ktore potrebuju etcd klienta
sa registruju neskor a dobehnu ako prve,a az potom sa ukonci etcd spojenie.

Tak to bolo aj myslene, ale nedoslo mi, ze ten hlavny kontext processu to zrusi skor.